public/hooks/use_chat_state.test.tsx

```diff
@@ -0,0 +1,193 @@
+    act(() => {
+      result.current.chatStateDispatch({
+        type: 'send',
+        payload: { type: 'input', contentType: 'text', content: 'question mock' },
+      });
+    });
+    expect(result.current.chatState.llmResponding).toBe(true);
+    expect(result.current.chatState.messages).toEqual([
+      { type: 'input', contentType: 'text', content: 'question mock' },
+    ]);
+  });
+
+  it('should reset to initial state', () => {
+    const { result } = renderHook(() => useChatState(), { wrapper: ChatStateProvider });
+    act(() => {
+      result.current.chatStateDispatch({
+        type: 'send',
+        payload: { type: 'input', contentType: 'text', content: 'question mock' },
+      });
+    });
+    expect(result.current.chatState.llmResponding).toBe(true);
+    expect(result.current.chatState.messages).toEqual([
+      { type: 'input', contentType: 'text', content: 'question mock' },
+    ]);
+
+    act(() => {
+      result.current.chatStateDispatch({
+        type: 'reset',
+      });
+    });
+    expect(result.current.chatState).toEqual({
+      interactions: [],
+      messages: [],
+      llmResponding: false,
+    });
+  });
+
+  it('should update state after `receive`', () => {
+    const { result } = renderHook(() => useChatState(), { wrapper: ChatStateProvider });
+    act(() => {
+      result.current.chatStateDispatch({
+        type: 'send',
+        payload: { type: 'input', contentType: 'text', content: 'question mock' },
+      });
+    });
+    expect(result.current.chatState.llmResponding).toBe(true);
+    expect(result.current.chatState.messages).toEqual([
+      { type: 'input', contentType: 'text', content: 'question mock' },
+    ]);
+    expect(result.current.chatState.interactions).toEqual([]);
+
+    act(() =>
+      result.current.chatStateDispatch({
+        type: 'receive',
+        payload: {
+          messages: [
+            { type: 'input', contentType: 'text', content: 'question mock' },
+            { type: 'output', contentType: 'markdown', content: 'output mock' },
+          ],
+          interactions: [
+            {
+              input: 'question mock',
+              response: 'output mock',
+              conversation_id: 'conversation_id_mock',
+              interaction_id: 'interaction_id_mock',
+              create_time: new Date().toLocaleString(),
+            },
+          ],
+        },
+      })
+    );
+
+    expect(result.current.chatState.llmResponding).toBe(false);
+    expect(result.current.chatState.messages).toEqual([
+      { type: 'input', contentType: 'text', content: 'question mock' },
+      { type: 'output', contentType: 'markdown', content: 'output mock' },
+    ]);
+    expect(result.current.chatState.interactions).toEqual([
+      {
+        input: 'question mock',
+        response: 'output mock',
+        conversation_id: 'conversation_id_mock',
+        interaction_id: 'interaction_id_mock',
+        create_time: new Date().toLocaleString(),
```

Comment:
Shall we store the create time? Just wondering if the create time will be different with original one.